akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/AccountExampleOptionState.scala

```diff
@@ -0,0 +1,103 @@
+package docs.akka.persistence.typed
+
+import akka.actor.typed.Behavior
+import akka.persistence.typed.scaladsl.{Effect, PersistentBehaviors6}
+import akka.persistence.typed.scaladsl.PersistentBehaviors6.HandlerFactoryOption
+import docs.akka.persistence.typed.AccountExampleOptionStateWithFactory.AccountEntity.{CommandHandler, EventHandler}
+
+class AccountExampleOptionState {
+
+    object AccountEntity {
+
+      sealed trait AccountCommand
+      case object CreateAccount extends AccountCommand
+      case class Deposit(amount: Double) extends AccountCommand
+      case class Withdraw(amount: Double) extends AccountCommand
+      case object CloseAccount extends AccountCommand
+
+      sealed trait AccountEvent
+      case object AccountCreated extends AccountEvent
+      case class Deposited(amount: Double) extends AccountEvent
+      case class Withdrawn(amount: Double) extends AccountEvent
+      case object AccountClosed extends AccountEvent
+
+      type Effect = akka.persistence.typed.scaladsl.Effect[AccountEvent, Option[Account]]
+
+      sealed trait Account {
+        def applyCommand(cmd: AccountCommand): Effect
+        def applyEvent(event: AccountEvent): Account
+      }
+
+
+      case class OpenedAccount(balance: Double) extends Account {
+
+        override def applyCommand(cmd: AccountCommand): Effect =
+          cmd match {
+            case Deposit(amount) ⇒ Effect.persist(Deposited(amount))
+
+            case Withdraw(amount) ⇒
+              if ((this.balance - amount) < 0.0)
+                Effect.unhandled // TODO replies are missing in this example
+              else {
+                Effect
+                  .persist(Withdrawn(amount))
+                  .thenRun {
+                    case Some(OpenedAccount(0.0)) ⇒
```

Comment:
Here we can't avoid the Option to leak. The Behavior, and therefore the Effect, is typed to Option[Account]. 

I find this an annoyance. At that point in the evolution of that model there is no possibility that this is a None, still, the user has to unwrap it.